### PR TITLE
Single client instance

### PR DESCRIPTION
**Pull request by konraddysput**
# Why

Right now each client implements how to deal with the Backtrace "instance". The code is redundant. In addition to that storing the instance in a library "instance" instead of globally for every other instance might create some weird issues - for example: The error boundary won't find an instance. 

This diff fixes the behavior and forces all clients to use one single instance.

### CODE REVIEW

File: packages/react/src/BacktraceClient.ts [hunk 1]
```diff
@@ -43,21 +41,24 @@ export class BacktraceClient extends BrowserClient {
...
     }
 
     /**
      * Returns created BacktraceClient instance if the instance exists.
      * Otherwise undefined.
      */
     public static get instance(): BacktraceClient | undefined {
-        return this._instance;
+        return this._instance as BacktraceClient;
     }
```
Comment (perf2711) [inline packages/react/src/BacktraceClient.ts:63]: You should be able to move this to `BacktraceCoreClient` and mark it to return `this | undefined`.

Comment (konraddysput) [reply packages/react/src/BacktraceClient.ts:63]: goood idea.

Comment (konraddysput) [reply packages/react/src/BacktraceClient.ts:63]: I checked your idea and there is a problem with the instance variable type - we use it to make sure client is a singleton type. In this situation, we cannot type it to `this` 

image

**Review by perf2711 — APPROVED** [review]

File: packages/browser/src/BacktraceClient.ts [hunk 2]
```diff
@@ -59,21 +58,21 @@ export class BacktraceClient extends BacktraceCoreClient {
         options: BacktraceConfiguration,
         build?: (builder: BacktraceClientBuilder) => void,
     ): BacktraceClient {
-        if (this._instance) {
-            return this._instance;
+        if (this.instance) {
```
[the remark this answers is not in the document]
Comment (konraddysput) [reply packages/browser/src/BacktraceClient.ts:61]: No. Getter already cover casting the instance to the right type

[the remark this answers is not in the document]
Comment (adamcronin42) [reply packages/browser/src/BacktraceClient.ts:61]: Ohhhh. Forgot about the getter. Thanks!

**Review by adamcronin42 — APPROVED** [review]

**Pull request merged**